src/commands/up/app_source.rs

```diff
@@ -13,6 +13,7 @@ use spin_manifest::schema::v2::AppManifest;
 pub enum AppSource {
     File(PathBuf),
     OciRegistry(String),
+    BareWasm(PathBuf),
```

Comment:
Yes, I had wondered about this.  I'm concerned it would make the docs a bit confusing, given that `--from-file` (in the current proposal) also accepts Wasm files (because my lawyers inform me that Wasm files are files within the meaning of the Act).  I suppose we could deprecate `--from-file` and define a new `--from-manifest` flag that forces the file to be interpreted as TOML, but this feels excessive.  Although maybe I'll feel differently if I understand what #2529 is about.

Reply:
Maybe it would be better to instead have an orthogonal option to differentiate between manifestful and manifestless sources. Anyway, that ought to be backward compatible with this so definitely not a blocker here.